frontend/src/metabase/visualizations/visualizations/LinkViz/LinkViz.unit.spec.tsx

```diff
@@ -217,6 +218,14 @@ describe("LinkViz", () => {
       const searchInput = screen.getByPlaceholderText("https://example.com");
 
       userEvent.click(searchInput);
+      // There's a race here: as soon the search input is clicked into the text "Loading..."
+      // appears, and by the time we look for "Question Uno" the loading text may or may not
+      // still be there. Therefore, we wait for it if it's there. If it's not there it means
+      // that it's already come and gone, so we can swallow the error.
+      const doNothing = x => x;
+      await waitForElementToBeRemoved(() => screen.queryByText("Loading..."))
+        .then(doNothing)
```

Comment:
This is upsetting to me, but the linter yelled at me for not having a `then`.

Very open to alternatives to my little `doNothing` hack, surely there's a better idiom for this.

Reply:
(PS: the `catch` is definitely necessary: `waitForElementToBeRemoved` throws an error if it doesn't find the thing.)

Reply:
@tsmacdonald I've tried removing both `then` and `catch` locally, and seems like eslint is okay with it. FWIW, believe we never do things like that with `waitForElementToBeRemoved`, mind removing it?

Also, does the test really run into a `catch` block here on some runs? If so, that's an odd signal IMO

Reply:
First of all, thank you @tsmacdonald for trying to address any flaky tests, they're really preventing us from shipping, especially in this release week.

I wonder how you determine this is a race condition? I don't think unit tests should behave non-deterministically, although I realize that's not the case. I want to push this back, since what `waitForElementToBeRemoved` is doing is essentially waiting for some stuff to disappear with a default timeout of 1 second. That essentially means it should be almost equivalent to adding 1 second to the existing 1 second for `screen.findByText("Question Uno")` i.e. `screen.findByText("Question Uno", { timeout: 2000 })` (in case the first `waitForElementToBeRemoved` took the whole 1 second which is unlikely, but this is the upper bound)

However, since this is a unit test it shouldn't take that long to get an API response. The part where it waits for `Question Uno` to appear takes exactly 400ms (within a 5ms margin of error) from my testing. I know this used to be the case with `nock` before as Alex P. mentioned sometimes it doesn't even return any response at all, so no amount of timeout would help in that case.

In this case however, we seem to already be using the new `fetch-mock` library, so I'm not totally sure what's the deal with it in this case.

If you think you need to add `waitForElementToBeRemoved` still I'd suggest you try adding a timeout for `screen.findByText("Question Uno", { timeout: 2000 })` instead, which would be more straightforward. But again, we aim to not have to use any `timeout` at all.

Reply:
Discussed in Slack - WDYT about something without timeout, e.g.
```
userEvent.click(searchInput);
expect(await screen.findByText("Loading...")).toBeInTheDocument();
userEvent.click(await screen.findByText("Question Uno"));
```